### PR TITLE
update pre-commit, add gha, remove circleci

### PR DESCRIPTION
**Pull request by ralren**
## [Transition remaining GitHub repos from CircleCi to GHA](https://trello.com/c/at5KLKxa)

Changes proposed in this pull request:

- just remove the circleci folder
- add validate gha
- update pre-commit
- remove local hooks from pre-commit config

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -19,26 +19,3 @@ repos:
...
-        language: system
-        entry: poetry run black
-        types: [python]
-
-      - id: flake8
-        name: flake8
-        stages: [commit]
-        language: system
-        entry: poetry run flake8
-        types: [python]
```
Comment (rpdelaney) [inline .pre-commit-config.yaml]: Why this?

**Review by rpdelaney — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ralren) [reply .pre-commit-config.yaml]: the current validate gha doesn't know how to handle this

Comment (rpdelaney) [reply .pre-commit-config.yaml]: OK. I think we should leave this in and let GHA be broken until we fix it. Removing this stuff takes away pretty much all the linting.

Comment (ralren) [reply .pre-commit-config.yaml]: got it

**Review by rpdelaney — APPROVED** [review]
Thanks again for doing this!

**Pull request merged**